flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/source/config/SourceConfigConstants.java

```diff
@@ -0,0 +1,369 @@
+            "flink.stream.registerstreamconsumer.maxretries";
+
+    public static final int DEFAULT_REGISTER_STREAM_CONSUMER_RETRIES = 10;
+
+    /**
+     * The maximum time in seconds to wait for a stream consumer to become active before giving up.
+     */
+    public static final String REGISTER_STREAM_CONSUMER_TIMEOUT_SECONDS =
+            "flink.stream.registerstreamconsumer.timeout";
+
+    public static final Duration DEFAULT_REGISTER_STREAM_CONSUMER_TIMEOUT = Duration.ofSeconds(60);
+
+    /** The base backoff time between each registerStreamConsumer attempt. */
+    public static final String REGISTER_STREAM_CONSUMER_BACKOFF_BASE =
+            "flink.stream.registerstreamconsumer.backoff.base";
+
+    public static final long DEFAULT_REGISTER_STREAM_CONSUMER_BACKOFF_BASE = 500L;
+
+    /** The maximum backoff time between each registerStreamConsumer attempt. */
+    public static final String REGISTER_STREAM_CONSUMER_BACKOFF_MAX =
+            "flink.stream.registerstreamconsumer.backoff.max";
+
+    public static final long DEFAULT_REGISTER_STREAM_CONSUMER_BACKOFF_MAX = 2000L;
+
+    /** The power constant for exponential backoff between each registerStreamConsumer attempt. */
+    public static final String REGISTER_STREAM_BACKOFF_EXPONENTIAL_CONSTANT =
+            "flink.stream.registerstreamconsumer.backoff.expconst";
+
+    public static final double DEFAULT_REGISTER_STREAM_BACKOFF_EXPONENTIAL_CONSTANT = 1.5;
+
+    /**
+     * The maximum number of deregisterStreamConsumer attempts if we get a recoverable exception.
+     */
+    public static final String DEREGISTER_STREAM_CONSUMER_RETRIES =
+            "flink.stream.deregisterstreamconsumer.maxretries";
+
+    public static final int DEFAULT_DEREGISTER_STREAM_CONSUMER_RETRIES = 10;
+
+    /** The maximum time in seconds to wait for a stream consumer to deregister before giving up. */
+    public static final String DEREGISTER_STREAM_CONSUMER_TIMEOUT_SECONDS =
+            "flink.stream.deregisterstreamconsumer.timeout";
+
+    public static final Duration DEFAULT_DEREGISTER_STREAM_CONSUMER_TIMEOUT =
+            Duration.ofSeconds(60);
+
+    /** The base backoff time between each deregisterStreamConsumer attempt. */
+    public static final String DEREGISTER_STREAM_BACKOFF_BASE =
+            "flink.stream.deregisterstreamconsumer.backoff.base";
+
+    public static final long DEFAULT_DEREGISTER_STREAM_CONSUMER_BACKOFF_BASE = 500L;
+
+    /** The maximum backoff time between each deregisterStreamConsumer attempt. */
+    public static final String DEREGISTER_STREAM_CONSUMER_BACKOFF_MAX =
+            "flink.stream.deregisterstreamconsumer.backoff.max";
+
+    public static final long DEFAULT_DEREGISTER_STREAM_CONSUMER_BACKOFF_MAX = 2000L;
+
+    /** The power constant for exponential backoff between each deregisterStreamConsumer attempt. */
+    public static final String DEREGISTER_STREAM_BACKOFF_EXPONENTIAL_CONSTANT =
```

Comment:
`DEREGISTER_STREAM_BACKOFF_EXPONENTIAL_CONSTANT` -> `DEREGISTER_STREAM_CONSUMER_BACKOFF_EXPONENTIAL_CONSTANT`

Reply:
same

Reply:
addressed in previous comment